contrib/src/main/java/com/datatorrent/contrib/avro/AvroToPojo.java

```diff
@@ -0,0 +1,428 @@
+
+import com.google.common.collect.Lists;
+
+import com.datatorrent.api.AutoMetric;
+import com.datatorrent.api.Context;
+import com.datatorrent.api.Context.OperatorContext;
+import com.datatorrent.api.Context.PortContext;
+import com.datatorrent.api.DefaultInputPort;
+import com.datatorrent.api.DefaultOutputPort;
+import com.datatorrent.api.Operator;
+import com.datatorrent.api.annotation.InputPortFieldAnnotation;
+import com.datatorrent.api.annotation.OutputPortFieldAnnotation;
+import com.datatorrent.lib.util.FieldInfo;
+import com.datatorrent.lib.util.FieldInfo.SupportType;
+import com.datatorrent.lib.util.PojoUtils;
+
+/**
+ * <p>
+ * AvroToPojo
+ * </p>
+ * A generic implementation for conversion from Avro to POJO. The POJO class
+ * name & field mapping should be provided by the user. If this mapping is not
+ * provided then reflection is used to determine this mapping. As of now only
+ * primitive types are supported.
+ *
+ * @displayName Avro To Pojo
+ * @category Converter
+ * @tags avro
+ * @since 3.3.0
+ */
+
+@InterfaceStability.Evolving
+public class AvroToPojo implements Operator
+{
+
+  private List<Field> columnNames;
+
+  private Class<?> cls;
+
+  private static final String FIELD_SEPARATOR = ":";
+  private static final String RECORD_SEPARATOR = ",";
+
+  private String genericRecordToPOJOFieldsMapping = null;
+
+  public String getGenericRecordToPOJOFieldsMapping()
+  {
+    return genericRecordToPOJOFieldsMapping;
+  }
+
+  public void setGenericRecordToPOJOFieldsMapping(String genericRecordToPOJOFieldsMapping)
+  {
+    this.genericRecordToPOJOFieldsMapping = genericRecordToPOJOFieldsMapping;
+  }
+
+  private String schemaFile;
+
+  private Schema schema;
+
+  private List<FieldInfo> fieldInfos;
+
+  private transient List<ActiveFieldInfo> columnFieldSetters = null;
+
+  @AutoMetric
+  int recordCnt = 0;
+
+  @AutoMetric
+  int errorCnt = 0;
+
+  @AutoMetric
+  int fieldErrorCnt = 0;
+
+  public void parseSchema() throws IOException
+  {
+    setSchema(new Schema.Parser().parse(getSchemaString()));
+  }
+
+  public String getSchemaString()
+  {
+    return schemaFile;
+  }
+
+  public void setSchemaString(String schemaFile)
+  {
+    this.schemaFile = schemaFile;
+  }
+
+  public Schema getSchema()
+  {
+    return schema;
+  }
+
+  public void setSchema(Schema schema)
+  {
+    this.schema = schema;
+  }
+
+  public byte[] serialize(Object obj) throws IOException
+  {
+    try (ByteArrayOutputStream b = new ByteArrayOutputStream()) {
+      try (ObjectOutputStream o = new ObjectOutputStream(b)) {
+        o.writeObject(obj);
+      }
+      return b.toByteArray();
+    }
+  }
+
+  @InputPortFieldAnnotation(optional = false)
+  public final transient DefaultInputPort<GenericRecord> data = new DefaultInputPort<GenericRecord>()
+  {
+
+    @Override
+    public void process(GenericRecord tuple)
+    {
+
+      try {
+        Object obj = getPOJOFromGenericRecord(tuple, getCls());
```

Comment:
Please move this code from process method at the class level and then call that method from here.
This is so that if one wants to extend this, he/she can override the process functionality too.